pillar/fluentd/server_operations_qa.sls

```diff
@@ -1,38 +1,30 @@
-{% set app_name = "fluentd-aggregators" %}
-{% set fluentd_shared_key = salt.vault.read('secret-operations/operations-qa/fluentd_shared_key').data.value %}
-{% set heroku_http_token = salt.vault.read('secret-operations/operations-qa/heroku_http_token').data.value %}
-{% set mailgun_webhooks_token = salt.vault.read('secret-operations/operations-qa/mailgun_webhooks_token').data.value %}
-{% set odl_wildcard_cert = salt.vault.read('secret-operations/global/odl_wildcard_cert') %}
-{% import_yaml 'fluentd/fluentd_directories.yml' as fluentd_directories %}
-
-schedule:
-  refresh_{{ app_name }}_configs:
-    # Needed to ensure that S3 credentials remain valid
-    days: 5
-    function: state.sls
-    args:
-      - fluentd.config
+{% set ENVIRONMENT = salt.grains.get('environment') %}
+{% set mailgun_webhooks_token = salt.vault.read('secret-operations/{}/mailgun_webhooks_token'.format(ENVIRONMENT)).data.value %}
+{% set es_hosts = 'operations-elasticsearch.query.consul' %}
+{% set cert = salt.vault.cached_write('pki-intermediate-operations/issue/fluentd-server', common_name='{}'.format(es_hosts)) %}
 
 fluentd:
-  persistent_directories: {{ fluentd_directories|tojson }}
   overrides:
     nginx_config:
       server_name: logs-qa.odl.mit.edu
       cert_file: log-input.crt
       key_file: log-input.key
       cert_contents: __vault__::secret-operations/global/odl_wildcard_cert>data>value
       key_contents: __vault__::secret-operations/global/odl_wildcard_cert>data>key
+  cert:
+    - fluentd_cert: {{ cert.data.certificate }}
+    - fluentd_key: {{ cert.data.private_key }}
+    - ca_cert: {{ cert.data.issuing_ca }}
   plugins:
     - fluent-plugin-heroku-syslog-http
-    - fluent-plugin-s3
     - fluent-plugin-elasticsearch
   proxied_plugins:
     - route: heroku-http
       port: 9000
-      token: {{ heroku_http_token }}
+      token: __vault__::secret-operations/{{ ENVIRONMENT }}/heroku_http_token>data>value
```

Comment:
Recognizing that these values are already set, should we make this a `gen_if_missing`?

Reply:
Think we need to have it already set as we need to pass that token when we create the drain for that heroku app